meson_options.txt

```diff
@@ -0,0 +1,219 @@
+# Nix project build options
+#============================================================================
+
+# dirs
+#============================================================================
+
+option(
+    'nixstoredir',
+    type : 'string',
+    value : '/nix/store',
+    description : 'path of the Nix store (defaults to /nix/store)')
+
+
+
+
+# compiler args
+#============================================================================
+
+option(
+    'ldflags',
+    type : 'array',
+    value : [
+        '-L/usr/local/lib',
+        '-L/usr/lib',
+        '-L/lib'],
+    description : 'Link flags')
+
+
+option(
+    'cxxflags',
+    type : 'array',
+    value : [
+        '-Wno-pedantic',
+        '-Wno-non-virtual-dtor',
+        '-Wno-unused-parameter',
+        '-Wno-variadic-macros',
+        '-Wdeprecated-declarations',
+        '-Wno-missing-field-initializers'],
+    description : 'C build flags')
```

Comment:
rather than doing this, and fighting with the builtin $CXXFLAGS and `-Dcpp_args` handling, why not add a snippet like:
```meson
add_project_arguments(
  cpp.get_supported_arguments([
    '-Wno-pedantic',
    '-Wno-non-virtual-dtor',
    '-Wno-unused-parameter',
    ...,
  ]),
  language : 'cpp')
```

To the Meson.build files?

Reply:
fair point :)

Reply:
i'm looking at this again, and this is actually already in the code. all that is done,
is the decleration of the args is set in the options file, allowing them to be
overridden at build time.

https://github.com/p01arst0rm/nix/blob/0c6bcc3a3c256e7aba15c56b3decda168b5fd0b1/meson.build#L31